Panic when template has empty parameters

The following template results in a panic:
```
apiVersion: template.openshift.io/v1
kind: Template
objects:
- apiVersion: v1
  kind: Route
  metadata:
    labels:
      app: foo
    name: foo
  spec:
    host: foo.domain.com
    tls:
      insecureEdgeTerminationPolicy: Redirect
      termination: edge
    to:
      kind: Service
      name: foo
      weight: 100
    wildcardPolicy: None
parameters:
```

```
+ tailor --non-interactive -n foo apply
panic: interface conversion: interface {} is nil, not []interface {}
goroutine 34 [running]:
github.com/opendevstack/tailor/pkg/openshift.templateContainsTailorNamespaceParam(0xc000139120, 0xe, 0x2, 0x2, 0x3)
	pkg/openshift/template.go:102 +0x649
github.com/opendevstack/tailor/pkg/openshift.ProcessTemplate(0x7790e5, 0x1, 0xc000139102, 0xc, 0x7790e5, 0x1, 0xc000290000, 0x7f7d6ceeec18, 0xc00028c160, 0x0, ...)
	pkg/openshift/template.go:30 +0x2d1
github.com/opendevstack/tailor/pkg/commands.assembleTemplateBasedResourceList(0xc0002c6000, 0xc000290000, 0x7f7d6ceeec18, 0xc00028c160, 0xc00028c160, 0x0, 0x0)
	pkg/commands/diff.go:213 +0x4b3
github.com/opendevstack/tailor/pkg/commands.calculateChangeset.func1(0x0, 0x0)
	pkg/commands/diff.go:66 +0x8a
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000284300, 0xc000284330)
	pkg/mod/golang.org/x/sync@v0.0.0-20200625203802-6e8e738ad208/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
	pkg/mod/golang.org/x/sync@v0.0.0-20200625203802-6e8e738ad208/errgroup/errgroup.go:54 +0x66
```

--- a/internal/test/fixtures/template-param-detection/template-blank-parameters.yml
+++ b/internal/test/fixtures/template-param-detection/template-blank-parameters.yml
@@ -1,0 +1,20 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+objects:
+- apiVersion: v1
+  kind: Route
+  metadata:
+    labels:
+      app: foo
+    name: foo
+  spec:
+    host: foo.domain.com
+    tls:
+      insecureEdgeTerminationPolicy: Redirect
+      termination: edge
+    to:
+      kind: Service
+      name: foo
+      weight: 100
+    wildcardPolicy: None
+parameters:

--- a/pkg/openshift/template.go
+++ b/pkg/openshift/template.go
@@ -94,12 +94,12 @@ func templateContainsTailorNamespaceParam(filename string) (bool, error) {
 	default:
 		return false, errors.New("Not a valid template. Please see https://github.com/opendevstack/tailor#template-authoring")
 	}
-	objectsPointer, _ := gojsonpointer.NewJsonPointer("/parameters")
-	items, _, err := objectsPointer.Get(m)
-	if err != nil {
+	parametersPointer, _ := gojsonpointer.NewJsonPointer("/parameters")
+	parameters, _, err := parametersPointer.Get(m)
+	if err != nil || parameters == nil {
 		return false, nil
 	}
-	for _, v := range items.([]interface{}) {
+	for _, v := range parameters.([]interface{}) {
 		nameVal := v.(map[string]interface{})["name"]
 		if nameVal == nil {
 			return false, errors.New("Template parameter without 'name' property found")

--- a/pkg/openshift/template_test.go
+++ b/pkg/openshift/template_test.go
@@ -30,6 +30,11 @@ func TestTemplateContainsTailorNamespaceParam(t *testing.T) {
 			wantContains: false,
 			wantError:    "Not a valid template. Did you forget to add the template header?\n\napiVersion: v1\nkind: Template\nobjects: [...]",
 		},
+		"template with blank parameters": {
+			filename:     "template-blank-parameters.yml",
+			wantContains: false,
+			wantError:    "",
+		},
 		"garbage": {
 			filename:     "garbage.yml",
 			wantContains: false,

